cumulus_library/databases.py

```diff
@@ -238,22 +242,24 @@ def __init__(self, db_file: str):
         )
 
     def insert_tables(self, tables: dict[str, pyarrow.Table]) -> None:
-        """Ingests all ndjson data from a folder tree (often the output folder of Cumulus ETL)"""
+        """Ingests all ndjson data from a folder tree.
+
+        This is often the output folder of Cumulus ETL"""
         for name, table in tables.items():
             self.connection.register(name, table)
 
     @staticmethod
     def _compat_array_join(
-        value: Optional[list[Optional[str]]], delimiter: str
-    ) -> Optional[str]:
+        value: list[str | None] | None, delimiter: str
+    ) -> str | None:
```

Comment:
It can be `None` alright?! Geeze...

Reply:
That wouldn't help here exactly, because these were explicit Optionals not implicit.

But I can _see_ the logic for disallowing implicit optional - like if change the default value to not be `None` anymore, it's a little surprising for that to also change the type of the arg itself. But that's the kind of argument for a library with a strict API and not just devs making code.

I'm fine either way. Disabling individual rules is a little annoying, config wise, but maybe it's a win overall.